### DISCUSSION TITLE
Installing on MacOS

### DISCUSSION THREAD

**Discussion by ZSciamma** [InteractiveComputerGraphics/PositionBasedDynamics#121] [Q&A]
Hello,

I was wondering if installing this project works on MacOS. My configuration is:

MacOS Catalina 64-bit, CMake 3.22.1

I am new to CMake, but I tried to compile it using `cmake .` then `make`. Unfortunately, I got this error:

>[ 10%] Building CXX object discregrid/CMakeFiles/Discregrid.dir/src/discrete_grid.cpp.o
In file included from [...]/PositionBasedDynamics/extern/Discregrid/src/Ext_Discregrid/discregrid/src/discrete_grid.cpp:1:
In file included from [...]/PositionBasedDynamics/extern/Discregrid/src/Ext_Discregrid/discregrid/include/Discregrid/discrete_grid.hpp:6:
In file included from [...]/PositionBasedDynamics/extern/eigen/Eigen/Dense:1:
[...]/PositionBasedDynamics/extern/eigen/Eigen/Core:266:10: fatal error: 
      'omp.h' file not found
#include <omp.h>
         ^~~~~~~
1 error generated.
make[5]: *** [discregrid/CMakeFiles/Discregrid.dir/src/discrete_grid.cpp.o] Error 1
make[4]: *** [discregrid/CMakeFiles/Discregrid.dir/all] Error 2
make[3]: *** [all] Error 2
make[2]: *** [extern/Discregrid/src/Ext_Discregrid-stamp/Ext_Discregrid-build] Error 2
make[1]: *** [CMakeFiles/Ext_Discregrid.dir/all] Error 2
make: *** [all] Error 2

I tried `brew install libomp`, which is suggested by a number of StackOverflow posts (like [this one](https://stackoverflow.com/q/49738961/6148373)), but unfortunately that hasn't fixed the error.

I noticed that one of the latest commits is called "added support for MacOS", but since Mac isn't listed as one of the tested configurations on the readme, I was wondering if this project does still work for MacOS, and if maybe there's just something simple I've missed for installing it.

Thank you!

**Comment by janbender** [comment]
The problem is that we don't have any Mac to test the library. Maybe some other Mac user could help here.

**Reply by dandingol03** [reply]
> Hello,
> 
> I was wondering if installing this project works on MacOS. My configuration is:
> 
> MacOS Catalina 64-bit, CMake 3.22.1
> 
> I am new to CMake, but I tried to compile it using `cmake .` then `make`. Unfortunately, I got this error:
> 
> > [ 10%] Building CXX object discregrid/CMakeFiles/Discregrid.dir/src/discrete_grid.cpp.o
> > In file included from [...]/PositionBasedDynamics/extern/Discregrid/src/Ext_Discregrid/discregrid/src/discrete_grid.cpp:1:
> > In file included from [...]/PositionBasedDynamics/extern/Discregrid/src/Ext_Discregrid/discregrid/include/Discregrid/discrete_grid.hpp:6:
> > In file included from [...]/PositionBasedDynamics/extern/eigen/Eigen/Dense:1:
> > [...]/PositionBasedDynamics/extern/eigen/Eigen/Core:266:10: fatal error:
> > 'omp.h' file not found
> > #include <omp.h>
> > ^~~~~~~
> > 1 error generated.
> > make[5]: *** [discregrid/CMakeFiles/Discregrid.dir/src/discrete_grid.cpp.o] Error 1
> > make[4]: *** [discregrid/CMakeFiles/Discregrid.dir/all] Error 2
> > make[3]: *** [all] Error 2
> > make[2]: *** [extern/Discregrid/src/Ext_Discregrid-stamp/Ext_Discregrid-build] Error 2
> > make[1]: *** [CMakeFiles/Ext_Discregrid.dir/all] Error 2
> > make: *** [all] Error 2
> 
> I tried `brew install libomp`, which is suggested by a number of StackOverflow posts (like [this one](https://stackoverflow.com/q/49738961/6148373)), but unfortunately that hasn't fixed the error.
> 
> I noticed that one of the latest commits is called "added support for MacOS", but since Mac isn't listed as one of the tested configurations on the readme, I was wondering if this project does still work for MacOS, and if maybe there's just something simple I've missed for installing it.
> 
> Thank you!

add the path of `omp.h` to the `target_include_directories` in cmakelist.txt
